Void Elves

ercarp opened the pull request:
<!--
A basic changelog, goes to patch notes of the next version, so should be as short and informative as possible.
-->
## Changelog:
- Added a portrait modifier (`being_void`) for **_Void Elves_**.
- Added decision `Embrace Ren'dorei Culture` decision, allowing a character of Highborn heritage culture and `being_void` trait to change their culture to Void Elf. This also changes any vassals, family, and vassal's family who are also of Highborn heritage culture and either have the trait, or have the Void religion.

<!--
If you need to explain something to testers. Otherwise, delete this part.
-->
# How to test:
1. Test the portrait modifier in-game by using the console command `add_trait creature_void_elf ID` on high/blood elves, or by generating random characters in the Ruler Designer using the temporary `void_elf_teaser` ethnicity.
2. Provide feedback on the look of the void elves and how they could be (reasonably) improved.
3. Test the decision

ercarp committed:
Void Elf Trait + Portrait Modifier

- Added creature_void_elf trait and a portrait modifier that gives anyone with the trait the appearance of a void elf.

ercarp committed:
Void Elf Test Ethnicity (Temporary)

- Added a void elf ethnicity for testing purposes.

zumbak04 committed:
Merge remote-tracking branch 'remotes/origin/nightelf-markings' into void-elf-portraits

zumbak04 committed:
Void Elf Markings

Added void elf marking in marking gene instead of their own gene.
(GDC)

Deuce20201 reviewed (approved):
Visually, they look very nice.

TheDrPibb1999 reviewed (approved):
I think they look great, eye shadow is def a nice feature. applied to trait to many elves and had no issues, good work :)

zumbak04 commented on localization/english/wc_traits_l_english.yml:
```diff
@@ -29,6 +29,9 @@
  trait_creature_high_elf_desc:0 "The high elves, or Quel'dorei, are the absolute opposite of their cousins: the night elves. The Quel'dorei are the descendants of those who served Azshara at the time of the first Well of Eternity. The high elves used magic for generations, absorbing it with their mothers' milk. Even among those lacking a natural talent for magic, high elves cannot live or even th ...
  trait_creature_high_elf_character_desc:0 "$trait_creature_high_elf_desc$"
+ trait_creature_void_elf:0 "Void Elf"
+ trait_creature_void_elf_desc:0 "PH"
+ trait_creature_void_elf_character_desc:0 "PH
```
**_PH_**

ercarp replied:
Fixed in 2a27ee85a5afc74638bac35f4850e33b5beb33c9 and added localization.

ercarp committed:
Merge branch 'master' into void-elf-portraits

ercarp committed:
Void Elf Localization + Re-added Race Trait

- Added localization for the Void Elf trait.
- Re-added the Void Elf trait as it had somehow disappeared from the branch.

zumbak04 committed:
Merge remote-tracking branch 'remotes/origin/nightelf-markings' into void-elf-portraits

# Conflicts:
#	localization/english/religion/wc_core_tenets_l_english.yml

zumbak04 commented on localization/english/wc_traits_l_english.yml:
```diff
@@ -29,6 +29,9 @@
  trait_creature_high_elf_desc:0 "The high elves, or Quel'dorei, are the absolute opposite of their cousins: the night elves. The Quel'dorei are the descendants of those who served Azshara at the time of the first Well of Eternity. The high elves used magic for generations, absorbing it with their mothers' milk. Even among those lacking a natural talent for magic, high elves cannot live or even th ...
  trait_creature_high_elf_character_desc:0 "$trait_creature_high_elf_desc$"
+ trait_creature_void_elf:0 "Void Elf"
+ trait_creature_void_elf_desc:0 "Void elves are a sect of former blood elves that infused themselves with the powers of the Void for the sake of defending Azeroth, all the while resisting the Void's call to madness. In Thalassian, they are called #italic ren'dorei#!, which means "children of the Void"."
```
There was talking in the dev chat. I feel it resolves everything.
```suggestion
 trait_creature_void_elf_desc:0 "Void elves are a sect of former elves that infused themselves with the powers of the Void for the sake of defending Azeroth, all the while resisting the Void's call to madness. In Thalassian, they are called #italic ren'dorei#!, which means "children of the Void"."
```

zumbak04 commented on localization/english/wc_traits_l_english.yml:
```diff
@@ -29,6 +29,9 @@
  trait_creature_high_elf_character_desc:0 "$trait_creature_high_elf_desc$"
+ trait_creature_void_elf:0 "Void Elf"
+ trait_creature_void_elf_desc:0 "Void elves are a sect of former blood elves that infused themselves with the powers of the Void for the sake of defending Azeroth, all the while resisting the Void's call to madness. In Thalassian, they are called #italic ren'dorei#!, which means "children of the Void"."
+ trait_creature_void_elf_character_desc:0 "$trait_creature_void_elf_desc$""
```
**FLAVOR**
```suggestion
 trait_creature_void_elf_character_desc:0 "[ROOT.GetCharacter.GetFirstNameNoTooltip] is a void elf. $trait_creature_void_elf_desc$""
```

TheDrPibb1999 reviewed (changes requested):
"the sake of defending Azeroth," i thought it was history the main reason Magister Umbric began the whole void excursion was to
 make sure the Sin'dori would never fall again to their enemies? then they were recruited by the alliance.

TheGroxEmpire commented on localization/english/wc_traits_l_english.yml:
```diff
@@ -29,6 +29,9 @@
  trait_creature_high_elf_desc:0 "The high elves, or Quel'dorei, are the absolute opposite of their cousins: the night elves. The Quel'dorei are the descendants of those who served Azshara at the time of the first Well of Eternity. The high elves used magic for generations, absorbing it with their mothers' milk. Even among those lacking a natural talent for magic, high elves cannot live or even th ...
  trait_creature_high_elf_character_desc:0 "$trait_creature_high_elf_desc$"
+ trait_creature_void_elf:0 "Void Elf"
+ trait_creature_void_elf_desc:0 "Void elves are a sect of former blood elves that infused themselves with the powers of the Void for the sake of defending Azeroth, all the while resisting the Void's call to madness. In Thalassian, they are called #italic ren'dorei#!, which means "children of the Void"."
```
Capitalized the "Children of the Void". I also agree with @TheDrPibb1999 and that the phrase "for the sake of defending Azeroth" does not really fit as Void Elf description. It is not really explicitly mentioned in the quest cited by Wowpedia (https://wowpedia.fandom.com/wiki/The_Void_Elves_Stand_Ready) and more of a general theme during BfA that they are defending Azeroth. In GoA setting this does not really fit as they might have different agendas.
```suggestion
 trait_creature_void_elf_desc:0 "Void elves are a sect of former blood elves that infused themselves with the powers of the Void, all the while resisting the Void's call to madness. In Thalassian, they are called #italic ren'dorei#!, which means "Children of the Void"."
```

ercarp replied:
Changed this in f3dda92d467c9ed2c7533a1de5071c9db081ec94.

CatWitchLily commented:
The portraits look stunning, though is it expected that a void elf created in the Ruler Designer using the temporary ethnicity lacks the racial trait?

ercarp committed:
Update localization/english/wc_traits_l_english.yml

Co-authored-by: zumbak04 <46384992+zumbak04@users.noreply.github.com>

ercarp committed:
Update localization/english/wc_traits_l_english.yml

Co-authored-by: zumbak04 <46384992+zumbak04@users.noreply.github.com>

ercarp committed:
Trait Description Update

- Changed the trait description a bit.

TheDrPibb1999 reviewed (approved):
Meant to approve this months ago, my fault. LGTM

Grekopithikos committed:
Merge branch 'master' into void-elf-portraits

Grekopithikos committed:
Ren'dorei

Grekopithikos committed:
Merge remote-tracking branch 'origin/dev' into void-elf-portraits

Grekopithikos committed:
no race trait needed

Grekopithikos committed:
Embrace void elf Culture decision

robmart commented on common/culture/cultures/wc_highborne.txt:
```diff
@@ -112,6 +112,42 @@ blood_elf = {
+	
+	coa_gfx = { high_elven_coa_gfx western_coa_gfx }
+	building_gfx = { mediterranean_building_gfx }
+	clothing_gfx = { creature_high_elf_gfx high_elven_clothing_gfx high_elven_hairstyles_gfx high_elven_beards_gfx no_headgear_gfx }
```
Please add `being_void_gfx`

robmart committed:
Merge branch 'dev' into void-elf-portraits

robmart committed:
Revamp decision a little bit

robmart committed:
Finalize the Void Elf decision and events

robmart committed:
Change requirements mildly

robmart committed:
Change portrait modifier trigger

edmund-h commented on common/decisions/wc_culture_decisions.txt:
```diff
@@ -0,0 +1,73 @@
+		}
+		NOT = { culture = culture:void_elf }
+		culture = { has_cultural_pillar = heritage_highborne }
+		has_trait = being_void
```
there is no way to become  a void being in the game right now so this decision cannot be taken in the normal course of the game without debug mode. this needs to be fixed, otherwise this PR needs to be shelved until infusions are done.

edmund-h commented on common/decisions/wc_culture_decisions.txt:
```diff
@@ -0,0 +1,73 @@
+			any_vassal_or_below = {
+				has_title = title:c_sunwell_plateau
+			}
+		}
```
the title requirements are too onerous. in game, the void elves dont even have a title. it's challenging enough to become a void being as it is. i think they should be at least a duke.

edmund-h reviewed (changes requested):
The requirements to this decision need to be changed. The requirement for the titles is too onerous and dont match game lore, and right now there's no way to get the `being_void` trait in normal gameplay. 

i feel that maybe a new religion needs to be added to reflect void elves with the EK synthesis doctrine. otherwise, void elves will be hated on the same level as undead are. this is a big gameplay problem and will push them in the opposite direction as in lore. unions should also be factoring in void elves like they do undead. 

the PR works, but it is not ready to ship and needs major consideration before that happens. i think this has to be pushed

edmund-h commented:
Ok so this is what myself and @Raged50 settled on with input from @Grekopithikos 

requirements for decision: 
- highborne cultural group
- mystic trait
- at least duke title tier
- the sin'dorei must exist or you have a maxed learning education

event chain: 
- you build an arcane observatory
- you make contact with a dark naaru
- you convert to kareshi religion and travel the void for 1000 subjective years
- you become a void being and return to your people with the dark naaru enslaved
- you establish the void elf culture

edmund-h commented:
this PR is superseded by https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2/pull/1384
